Libraries/builtin/Tests/test_copy.cpp

```diff
@@ -0,0 +1,115 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <gtest/gtest.h>
+#include <builtin/copy/Options.h>
+#include <builtin/copy/Driver.h>
+#include <libutil/Filesystem.h>
+#include <libutil/MemoryFilesystem.h>
+#include <process/Context.h>
+#include <process/MemoryContext.h>
+#include <plist/Format/Encoding.h>
+
+using builtin::copy::Driver;
+using builtin::copy::Options;
+using libutil::Filesystem;
+using libutil::MemoryFilesystem;
+
+static std::vector<uint8_t>
+Contents(std::string const &string)
+{
+    return std::vector<uint8_t>(string.begin(), string.end());
+}
+
+static process::MemoryContext
+Process(std::string const &executable, std::vector<std::string> const &arguments)
+{
+    return process::MemoryContext(
+        executable,
+        "/",
+        arguments,
+        std::unordered_map<std::string, std::string>(),
+        0,
+        0,
+        "root",
+        "wheel");
+}
+
+TEST(copy, Name)
+{
+    Driver driver;
+    EXPECT_EQ(driver.name(), "builtin-copy");
+}
+
+TEST(copy, CopyFiles)
+{
+    std::vector<uint8_t> contents;
+    MemoryFilesystem filesystem = MemoryFilesystem({
```

Comment:
Idea to cut down on unit test clutter:  'using namespace MemoryFilesystem::Entry' and update types as appropriate.